app/src/store/views/orderFormView.ts

```diff
@@ -229,6 +239,39 @@ export default class OrderFormView {
     this.addlOptionsVisible = !this.addlOptionsVisible;
   }
 
+  /** requests a quote for an order to obtain accurate fees */
+  async quoteOrder() {
+    runInAction(() => {
+      this.executionFee = 0;
+      this.worstChainFee = 0;
+    });
+    if (!this.isValid) return;
+
+    const minUnitsMatch = Math.floor(this.minChanSize / ONE_UNIT);
+    const satsPerKWeight = this._store.api.pool.satsPerVByteToKWeight(
+      this.maxBatchFeeRate,
+    );
+
+    const {
+      totalExecutionFeeSat,
+      worstCaseChainFeeSat,
+    } = await this._store.orderStore.quoteOrder(
+      this.amount,
+      this.perBlockFixedRate,
+      this.derivedDuration,
+      minUnitsMatch,
+      satsPerKWeight,
+    );
+
+    runInAction(() => {
+      this.executionFee = totalExecutionFeeSat;
+      this.worstChainFee = worstCaseChainFeeSat;
+    });
+  }
+
+  /** quote order at most once every 2 seconds when using this func  */
+  quoteOrderThrottled = debounce(this.quoteOrder, 2000);
```

Comment:
This tripped me up a bit. I changed some values and expected the quote to be updated. Was almost ready to enter yet another number when I noticed the number finally updated. Maybe I'm just too impatient.

If it's easy to implement (probably not worth doing if it requires a lot of effort), we could invalidate the quote fields (for example by showing a loading animation) until the next debounce run comes along. That way the user knows something's going to happen soon.

Reply:
ah, good point. I like the idea of displaying a loader. I can do that quickly.

Reply:
I added the loaders and decreased the debounce to 1 second. Thanks for the suggestion